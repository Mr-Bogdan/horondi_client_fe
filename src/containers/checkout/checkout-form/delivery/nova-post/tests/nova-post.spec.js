import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { useSelector } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import NovaPost from '../nova-post';

// import { LIGHT_THEME } from '../../../../../../configs';
// import { getFromLocalStorage } from '../../../../../../services/local-storage.service'

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../nova-post.styles', () => ({ useStyles: () => ({ Theme: 'lightMode' }) }));
jest.mock('react-redux');

useSelector.mockImplementation(() => ({
  deliveryLoading: true,
  cities: '',
  warehouses: ''
}));

const props = {
  isLightTheme: true,
  language: 1,
  setFieldValue: jest.fn(),
  errors: jest.fn(),
  touched: jest.fn(),
  values: {
    firstName: '',
    lastName: '',
    email: 'test@gmail.com',
    phoneNumber: '000'
  }
};

describe('NovaPost component tests', () => {
  it('Should render NovaPoshta', () => {
    const wrapper = shallow(<NovaPost {...props} />);
    expect(wrapper).toBeDefined();
  });
  it('Should 2', () => {
    const wrapper = shallow(<NovaPost {...props} />);
    const actual = wrapper.find('h3').text();
    const expected = 'Delivery Address';
    expect(actual).toEqual(expected);
  });
  it('should handle change', () => {
    const wrapper = shallow(<NovaPost {...props} />);
    expect(wrapper).toMatchSnapshot();

    // const changeEvent = { target: { value: 101 } };
    // expect(wrapper.find(props.setFieldValue)).to.have.lengthOf(1);
    // expect(props.setFieldValue).not.toBeCalled();
    // wrapper.simulate('change', changeEvent);
    // expect(props.setFieldValue).toBeCalledWith(changeEvent);
  });
});
