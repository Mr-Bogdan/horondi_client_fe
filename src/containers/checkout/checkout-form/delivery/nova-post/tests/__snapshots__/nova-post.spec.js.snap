// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NovaPost component tests should handle change 1`] = `
<div>
  <h3>
    Delivery Address
  </h3>
  <div>
    <div>
      <WithStyles(ForwardRef(Autocomplete))
        getOptionLabel={[Function]}
        noOptionsText="City not found"
        onChange={[Function]}
        onInputChange={[Function]}
        options=""
        renderInput={[Function]}
      />
    </div>
  </div>
  <div>
    <div>
      <WithStyles(ForwardRef(Autocomplete))
        disabled={true}
        getOptionLabel={[Function]}
        noOptionsText="Department not found"
        onChange={[Function]}
        onInputChange={[Function]}
        options={Array []}
        renderInput={[Function]}
      />
    </div>
  </div>
</div>
`;
